src/commands/logs/list.rs

```diff
@@ -143,10 +168,327 @@ fn execute_single_fetch(
     Ok(())
 }
 
+/// Manages deduplication of log entries with a bounded buffer
+struct LogDeduplicator {
+    /// Set of seen log IDs for quick lookup
+    seen_ids: HashSet<String>,
+    /// Buffer of log entries in order (for maintaining size limit)
+    buffer: VecDeque<LogEntry>,
+    /// Maximum size of the buffer
+    max_size: usize,
+}
+
+const MAX_DEDUP_BUFFER_SIZE: usize = 10_000;
+
+impl LogDeduplicator {
+    fn new(max_size: usize) -> Self {
+        Self {
+            seen_ids: HashSet::new(),
+            buffer: VecDeque::new(),
+            max_size,
+        }
+    }
+
+    /// Add new logs and return only the ones that haven't been seen before
+    fn add_logs(&mut self, new_logs: Vec<LogEntry>) -> Vec<LogEntry> {
+        let mut unique_logs = Vec::new();
+
+        for log in new_logs {
+            if !self.seen_ids.contains(&log.item_id) {
+                self.seen_ids.insert(log.item_id.clone());
+                self.buffer.push_back(log.clone());
+                unique_logs.push(log);
+            }
+        }
+
+        // Maintain buffer size limit by removing oldest entries
+        while self.buffer.len() > self.max_size {
+            if let Some(removed_log) = self.buffer.pop_front() {
+                self.seen_ids.remove(&removed_log.item_id);
+            }
+        }
+
+        unique_logs
+    }
+}
+
+/// Tracks consecutive batches of all-new logs and manages warning state.
+///
+/// A batch is "all-new" when every fetched log is unique (no duplicates).
+/// This struct tracks how many consecutive all-new batches we've seen and
+/// warns when the count reaches the threshold, suggesting the user might be
+/// missing some logs due to overly broad filtering.
```

Comment:
It is not clear to me why we want this warning, could you explain it? We could also discuss in a meeting so I can better understand the goal

Reply:
Does every log being unique mean that none of the logs have been seen in a previous fetch? If so, what is the expected behavior if the most recent fetch itself contains duplicated logs? Or, is this not possible, since every log (even if it is the same message) has the same ID?